### ISSUE TITLE
Multispeaker TTS in model zoo seems to have variable quality

### ISSUE THREAD

**Issue by picheny-nyu** [espnet/espnet#2764]
I have been synthesizing some TTS using the multi speaker TTS systems described in the notebooks and am getting some strange results. Sometimes the output speech resembles the speakers used to create the embeddings and sometimes they don't. Also sometimes the output is missing words or otherwise garbles part of the input text sentences. Is this to be expected? I am using:

tag = "kan-bayashi/vctk_gst_transformer"
vocoder_tag = "vctk_parallel_wavegan.v1"

I also tried the tacotron model, it is worse.  I attach zip files. Input_a and input_b are original files used to create embeddings; avalanche and congress are the texts:

 Avalanche: 'You saw an avalanche of positive comments coming from Republican senators on Thursday.’
 Congress: 'Congress is up against the clock, which means it is going to be a consequential and, yes, messy week on Capitol Hill.'

and the a and b versions are those corresponding to the target speakers (input_a and input_b). Anything obviously wrong here? I also attach the python code I used to generate this (maybe I am calling things incorrectly?)

[Archive.zip](https://github.com/espnet/espnet/files/5659708/Archive.zip)

#!/usr/bin/env python
# coding: utf-8

# ## Multi-speaker Model Demo

###################################
#          ENGLISH MODELS         #
###################################
fs, lang = 24000, "English"
# tag = "kan-bayashi/vctk_gst_tacotron2"
tag = "kan-bayashi/vctk_gst_transformer"
vocoder_tag = "vctk_parallel_wavegan.v1"
# vocoder_tag = "vctk_multi_band_melgan.v2"

import time
import torch
import sys

from espnet_model_zoo.downloader import ModelDownloader
from espnet2.bin.tts_inference import Text2Speech
from parallel_wavegan.utils import download_pretrained_model
from parallel_wavegan.utils import load_model

d = ModelDownloader()
text2speech = Text2Speech(
    **d.download_and_unpack(tag),
    device="cuda",
    # Only for Tacotron 2
    threshold=0.5,
    minlenratio=0.0,
    maxlenratio=10.0,
    use_att_constraint=False,
    backward_window=1,
    forward_window=3,
    # Only for FastSpeech & FastSpeech2
    speed_control_alpha=1.0,
)
text2speech.spc2wav = None  # Disable griffin-lim

# NOTE: Sometimes download is failed due to "Permission denied". That is 
#   the limitation of google drive. Please retry after serveral hours.

vocoder = load_model(download_pretrained_model(vocoder_tag)).to("cuda").eval()
vocoder.remove_weight_norm()

x = sys.argv[1]

import soundfile as sf
import librosa

speech, fs = sf.read(sys.argv[2])

if (fs != 24000) :
    speech = librosa.resample(speech, fs, 24000.)

speech = torch.from_numpy(speech).float()

# synthesis
with torch.no_grad():
    start = time.time()
    wav, c, *_ = text2speech(x, speech=speech)
    wav = vocoder.inference(c)
rtf = (time.time() - start) / (len(wav) / fs)
print(f"RTF = {rtf:5f}")

if (len(sys.argv) == 5): 
    sf.write(sys.argv[4], speech.view(-1).cpu().numpy(), 24000)
sf.write(sys.argv[3], wav.view(-1).cpu().numpy(), 24000)

**Comment by unilight** [comment]
The task you want to perform is GST is _non-parallel style transfer_, where the content of the input text and the reference signal are not the same.
GST is not so good at this task. In section 6.2.2 of the [original paper](https://arxiv.org/pdf/1803.09017), the authors only experimented that the attention could stay robust in such a non-parallel setting. But it did not mention the similarity. In fact, many researchers have focused on improving that, such as [this paper](https://arxiv.org/abs/1907.02479).

**Comment by picheny-nyu** [comment]
Thanks! So just to clarify, extracting an embedding from a speaker on a fixed text with no transcription and using that as input to a TTS system to make the TTS sound like that speaker for a variety of new text is still the issue of ongoing research in the TTS community, correct?

**Comment by unilight** [comment]
I think so. I don't know how far we are from solving it, but at least in this year's Interspeech I still saw several papers working on this topic, including this one: https://www.isca-speech.org/archive/Interspeech_2020/pdfs/1251.pdf

**Comment by Piteryo** [comment]
@unilight 
So you mean that all these experiments with Multispeaker in this repo will produce bad speech in a case when the target speech (for extracting speaker and GST information) doesn't match input text? Is it because of using GST for speaker information extraction and it could be fixed by switching to speaker embeddings (x-vector or smth else)? Or there is no stable Multispeaker model for TTS for now at all?

**Comment by unilight** [comment]
@Piteryo "Bad" speech is just in terms of speaker similarity. According to the samples provided by @picheny-nyu , the quality sounded okay. Also, you can check out the x-vector based multi-speaker TTS models, although I personally have never tested the performance with unseen speakers...

**Comment by Piteryo** [comment]
>  "Bad" speech is just in terms of speaker similarity. 

But OP says that 

>  Also sometimes the output is missing words or otherwise garbles part of the input text sentences. Is this to be expected?

I'm facing the same behaviour - the quality in terms of pronunciation heavily depends on the target sample. Also "speaker classification" thing works badly - sometimes it can produce output for the wrong speaker even when I use target sample from training data.

> Also, you can check out the x-vector based multi-speaker TTS models, although I personally have never tested the performance with unseen speakers...

But the performance for speakers who were in training data should be ok?

**Comment by unilight** [comment]
I think it is hard to say whether the misspelling and the missing words come from imperfect acoustic modeling or speaker modeling. Also, as there is hardly an objective metric for speaker identity, it is hard to say whether we successfully reproduced Google's model or not.

**Comment by Piteryo** [comment]
> I think it is hard to say whether the misspelling and the missing words come from imperfect acoustic modeling or speaker modeling

But burbling speech appears and disappears on the same text input with different target samples. So clearly the problem exists in speaker modelling. 

I'll try to teach x-vector model and reply here whether such type of issue exists there or not.

**Comment by kan-bayashi** [comment]
GST is not so good in terms of multi-speaker modeling but maybe we can slightly reduce the problem by trimming the audio not to contain the silence at the end of audio. Since the reference embedding is the last state of unidirectional GRU, the long silence at the end of audio will be a problem.

If you want to definitely improve the speaker similarity, we need to use another model:
e.g. https://arxiv.org/pdf/2005.08484.pdf

**Comment by kan-bayashi** [comment]
I add the support of X-vector in ESPnet2.
If you are interested, please check it. #2796

**Comment by kan-bayashi** [comment]
I confirmed that the use of x-vector is much stable than GST.
Also, they can be combined and the quality seems to be slightly improved.
https://drive.google.com/drive/folders/1Zv1HIPUEGKcBbGHF_ay6YIHtgp5CLYcv?usp=sharing

**Comment by inconnu11** [comment]
Hi, the repo provided the pretrained model [libritts_tts_train_xvector_conformer_fastspeech2_transformer_teacher_raw_phn_tacotron_g2p_en_no_space_train.loss ](https://zenodo.org/record/4418754). I inferenced on the LibriTTS using this pretrained ckpt. I see that you upload some samples of VCTK in google drive. Could you please upload some LibriTTS samples on train-set, dev-set and test-set for me to have a comparison?
